src/language/templates/ast/all.hpp

```diff
@@ -20,13 +20,25 @@
  */
 
 #include "ast/ast.hpp"
-{% for header in node_info.CIRCULAR_DEPENDECIES %}
-#include "ast/{{header}}.hpp"
-{% endfor %}
 
 {% for node in nodes %}
 #ifndef {{ node.cpp_fence }}
 #define {{ node.cpp_fence }}
+#define {{ node.cpp_fence }}_INLINE_DEFINITION_REQUIRED
```

Comment:
This could be added only `if node.has_template_methods`?

Reply:
Absolutely, there is no need to pollute with this macro if there is not inline definitions below.